### ISSUE TITLE
Powerline shows a "1" after the prompt

### ISSUE THREAD

**Issue by GoDzM4TT3O** [powerline/powerline#2069]
When I open a terminal, this is what I see:
![default][default]
The prompt is `matteo>~>`
But, when I type any command, a "1" appears after the prompt:
![error][error]
The expected result should be `matteo>~>`, but the actual result is `matteo>~>1>`
How do I fix this?

[default]: https://i.postimg.cc/pX3HCv5N/default.png
[error]: https://i.postimg.cc/Hk5CPNbr/error.png

**Comment by PH111P** [comment]
I suspect that the "1" that you see is the return value from the most recently run script; at least that should be the segment responsible for showing the "1".
While in theory, `echo "test"` should not return `1` (which should be investigated); a quick fix to your problem would be to remove the `powerline.segments.shell.last_pipe_status` module from your shell configuration.

**Comment by GoDzM4TT3O** [comment]
I use the standard powerline configuration. Where is `powerline.segments.shell.last_pipe_status` located? I installed powerline using pip, so my powerline directory is in `/home/matteo/.local/lib/python2.7/site-packages/powerline/`. I also noticed that, if I press enter without any command, that "1" goes away, unless I type any other command.

**Comment by GoDzM4TT3O** [comment]
I fixed this by modifiying `/home/matteo/.local/lib/python2.7/site-packages/powerline/bindings/bash/powerline.sh` after a bit of digging; I removed "last_pipe_status" and "jobnum" from $PS1, $PS2 and $PS3 and I also commented out jobnum and renderer-arg.

This actually makes it work. I don't know why your code does that but my simple fix made it work.

**Comment by PH111P** [comment]
> I fixed this by modifiying `/home/matteo/.local/lib/python2.7/site-packages/powerline/bindings/bash/powerline.sh` after a bit of digging; I removed "last_pipe_status" and "jobnum" from $PS1, $PS2 and $PS3 and I also commented out jobnum and renderer-arg.
> 
> This actually makes it work. I don't know why your code does that but my simple fix made it work.

Well, this "quick fix" is probably one of the dirtiest methods around (in particular it breaks when updating), so for the record, let me point you to the relevant section of the documentation on how to configure powerline in a _slightly_ less destructive way: https://powerline.readthedocs.io/en/latest/configuration.html

TL;DR: copy the default config files from `powerline_root/config_files` to `.config/powerline` and edit `.config/powerline/themes/shell/default.json`; in particular remove the `last_pipe_status` module from the json file.

**Comment by PH111P** [comment]
I added some more information to the Readme that should give more details. 
Closing this issue as it seems to be just a configuration problem.

**Comment by GoDzM4TT3O** [comment]
Reopening as I have installed a brand new version of Ubuntu 19.10, I followed your instructions and the problem still persists.

**Comment by PH111P** [comment]
Did you try restarting the `powerline-daemon` after the configuration change?

**Comment by GoDzM4TT3O** [comment]
Yes, I did multiple times.

**Comment by GoDzM4TT3O** [comment]
Here's what I did:

`git clone https://github.com/powerline/powerline`
`cd powerline/powerline/config_files`
`mkdir ~/.config/powerline`
`cp * ~/.config/powerline; cd !$`
`powerline-lint; powerline-daemon --replace`

(`cd !$` is used to go to the directory in the previous command)
There were no errors printed by `powerline-lint`. I also tried:
- [x] reinstalling powerline to no avail;
- [x] removing `last_pipe_status` from the official configuration file, yet nothing has worked.

**Comment by PH111P** [comment]
can you post your current powerline shell config (located at `~/.config/powerline/themes/shell/default.json`)?

**Comment by GoDzM4TT3O** [comment]
```json
{
	"segments": {
		"left": [
			{
				"function": "powerline.segments.shell.mode"
			},
			{
				"function": "powerline.segments.common.net.hostname",
				"priority": 10
			},
			{
				"function": "powerline.segments.common.env.user",
				"priority": 30
			},
			{
				"function": "powerline.segments.common.env.virtualenv",
				"priority": 50
			},
			{
				"function": "powerline.segments.shell.cwd",
				"priority": 10
			},
			{
				"function": "powerline.segments.shell.jobnum",
				"priority": 20
			}
		],
		"right": [
			{
				"function": "powerline.segments.common.vcs.stash",
				"priority": 50
			},
			{
				"function": "powerline.segments.common.vcs.branch",
				"priority": 40
			}
		]
	}
}
```
Here it is!

**Comment by BlitzKraft** [comment]
@GoDzM4TT3O 
The orange-ish `1` is `jobnum`. This is not the `last_pipe_status` which shows up in a darker red color in default colors. Removing the jobnum will remove the number. This shows the number of background jobs running, if any. 

This issue is not present in newer versions. I have come across this issue in some of the older releases. Try and install the latest release.

**Comment by GoDzM4TT3O** [comment]
Ok, thank you!!!! :D

**Comment by GoDzM4TT3O** [comment]
I tried building powerline myself, but that didn't work either: the 1 was still showing.

`cd $HOME`
`git clone https://github.com/powerline/powerline -b develop`
`cd powerline`
`python setup.py build; sudo setup.py install`
`cd powerline/config_files; powerline_lint`

After modifying the file where powerline should be started (it's somewhere under /usr), powerline is now slower (I think it's because everytime it needs to read from /usr), and, sadly, still shows that damned orange "1". I tried removing the jobnum from the config file, but that made `powerline-lint` spit out a bunch of errors.

**Comment by BlitzKraft** [comment]
Did you fully remove/uninstall powerline from the previous installation? powerline can be installed at user level using `pip install --user ./` from the powerline directory. 

The powerline lint might be due to a stray comma. 

```json
{
	"segments": {
		"left": [
			{
				"function": "powerline.segments.shell.mode"
			},
			{
				"function": "powerline.segments.common.net.hostname",
				"priority": 10
			},
			{
				"function": "powerline.segments.common.env.user",
				"priority": 30
			},
			{
				"function": "powerline.segments.common.env.virtualenv",
				"priority": 50
			},
			{
				"function": "powerline.segments.shell.cwd",
				"priority": 10
			}
		],
		"right": [
			{
				"function": "powerline.segments.common.vcs.stash",
				"priority": 50
			},
			{
				"function": "powerline.segments.common.vcs.branch",
				"priority": 40
			}
		]
	}
}
```

**Comment by GoDzM4TT3O** [comment]
It still shows up, but I think I might need to clean it up. What commands should I run in order to completely clean up powerline?

**Comment by GoDzM4TT3O** [comment]
EDIT: I ran
```bash
pip uninstall powerline-status
rm -rf ~/.config/powerline*
git clone https://github.com/powerline/powerline
cd powerline
pip install --user ./
mkdir ~/.config/powerline
cp -r ~/powerline/powerline/config_files ~/.config/powerline
cd ~/.config/powerline/themes/shell
```
I then removed `default.json` with the file that @BlitzKraft sent, but I added `last_pipe_status`. This configuration DOES NOT SHOW THE ORANGE 1 ANYMORE! It also works correctly when trying to run a loop in the shell (see image-1 below). Works normally after running any command (see image-2 below)
```json
{
	"segments": {
		"left": [
			{
				"function": "powerline.segments.shell.mode"
			},
			{
				"function": "powerline.segments.common.net.hostname",
				"priority": 10
			},
			{
				"function": "powerline.segments.common.env.user",
				"priority": 30
			},
			{
				"function": "powerline.segments.common.env.virtualenv",
				"priority": 50
			},
			{
				"function": "powerline.segments.shell.cwd",
				"priority": 10
			}
		],
		"right": [
			{
				"function": "powerline.segments.shell.last_pipe_status",
				"priority": 10
			},
			{
				"function": "powerline.segments.common.vcs.stash",
				"priority": 50
			},
			{
				"function": "powerline.segments.common.vcs.branch",
				"priority": 40
			}
		]
	}
}
```
image-1:
![image-1]

image-2:
![image-2]


[image-1]: https://i.ibb.co/r7KWMbg/Screenshot-from-2020-02-20-14-57-42.png
[image-2]: https://i.ibb.co/wg7rxPM/Screenshot-from-2020-02-20-14-58-31.png
